### ISSUE TITLE
Choose the proper form to evaluate

### ISSUE THREAD

**Issue by tomaskulich** [bhurlow/vim-parinfer#32]
searching opening '(' on the first line seems like a decent heuristics for the beginning of the form. 

```
  let topline = search('^(', 'bn') - 1
```

The bottomline however is searched for badly. Searching for corresponding ')' may not work, because there may not yet be corresponding closing bracket:

```
  let bottomline = searchpair('(','',')', 'n') + 1
```

Good idea may be to look for the next opening bracket on the first line (i.e. beginning of the next 'main' form) (or the end of file if not existing). I'm not sure how to implement this in vimscript though.

**Comment by bhurlow** [comment]
@tomaskulich yep, this is definitely the tricky part thanks for the bump. I think the topline search is adequate as this is what `vim-sexp` is doing. Maybe a fallback for `searchpair()` is the next step

**Comment by tomaskulich** [comment]
Well, if I understand correctly what the plugin is trying to do, then the `searchpair` approach is IMO conceptually broken; you cannot ask for matching paren, if the paren is yet-to-be inserted - which is very often the case. The `searchpair` will work correctly only if there is nothing to do (with this kind of parens)

**Comment by bhurlow** [comment]
@tomaskulich I'm thinking the best thing here might be like you proposed, too count parens until we hit the ***next*** opening `(`

**Comment by shaunlebron** [comment]
The proposed solution is what atom-parinfer does 👍

**Comment by bhurlow** [comment]
@shaunlebron good to know thanks!

**Comment by bhurlow** [comment]
ok, I did this: https://github.com/bhurlow/vim-parinfer/commit/8c5269b965b405c58b63a0154773c961e1eece8b. Could be few edge cases still left, although this definitely works smoother
